### PR TITLE
Implement dynamic server endpoints (#50)

### LINKED ISSUE #50: Allow SaltyRTC server endpoint to be dynamic

### PR DESCRIPTION
**Pull request by dbrgn**
@lgrahl @sillych what do you think?

Fixes #50.

### CODE REVIEW

**Comment by dbrgn** [comment]
Note: I chose not to throw any exceptions in the `getXxx` methods. Operations that might throw should be done in the constructor or outside the class.

File: docs/docs/usage.md [hunk 1]
```diff
@@ -103,6 +103,43 @@ final SaltyRTC client = new SaltyRTCBuilder()
         .asInitiator();
 ```
 
+## Trusted keys
+
+TODO
+
+## Dynamically determine server connection info
```
Comment (lgrahl) [inline docs/docs/usage.md]: http://www.titlecapitalization.com

Comment (dbrgn) [inline docs/docs/usage.md]: https://reviewediting.wordpress.com/2012/07/10/titles-sentence-style-or-headline-style/

> It’s also a good style to use if you have many long titles in your document.

I sometimes prefer that to the more complex rules of title capitalization depending on who you ask. This is the AP style:

> Capitalize the first word of every letter except articles, coordinating conjunctions, and prepositions of three letters or fewer. There’s one exception: Any word that is the first word in the headline or the last word should be capitalized, regardless of its part of speech.

And they call that "simple". WTF.

Anyways, feel free to fix the titles if they bother you.

Comment (lgrahl) [inline docs/docs/usage.md]: I'll do it if it triggers me enough at some point. :grin:

**Pull request merged**